frontend/src/logic/real-schedule-logic/schedule.logic.ts

```diff
@@ -137,18 +108,27 @@ export class ScheduleLogic {
   }
 
   public updateRow(row: DataRow) {
-    Object.values(this.sections).forEach((section) => {
-      section && section.tryUpdate(row);
-    });
+    this.nurseInfoProvider.tryUpdate(row);
+    this.babysitterInfoProvider.tryUpdate(row);
+    this.childrenInfoProvider.tryUpdate(row);
   }
 
-  public updateSection(section: keyof ScheduleSections, newSectionData: DataRow[]) {
+  //TODO: Check if it still needs to refactored
+  public updateChildrenSection(newSectionData: DataRow[]) {
     // Refactor
     let data = DataRowHelper.dataRowsAsValueDict<any>(newSectionData, true);
-    if (section === "childrenInfo") {
-      this.sections.childrenInfo = new ChildrenInfoLogic({ ...data });
-    } else {
-      this.sections[section] = new ShiftsInfoLogic({ ...data });
-    }
+    this.childrenInfoProvider = new ChildrenInfoLogic({ ...data });
+  }
+
+  public updateNurseSection(newSectionData: DataRow[]) {
+    // Refactor
+    let data = DataRowHelper.dataRowsAsValueDict<any>(newSectionData, true);
+    this.nurseInfoProvider = new ShiftsInfoLogic({ ...data });
+  }
+
+  public updateBabysitterSection(newSectionData: DataRow[]) {
+    // Refactor
```

Comment:
Co dokładnie chcesz zmienić w tym miejscu ?

Reply:
Bo dla mnie wygląda dobrze

Reply:
A to w porządku. Wcześniej był ten komentarz w tej jednej funkcji, która podzieliłem na 3 i właśnie nie widziałem o co chodzi. To usunę te komentarze